src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -707,12 +723,29 @@ void IrcMessageHandler::handleUserNoticeMessage(Communi::IrcMessage *message,
         else if (msgType == "announcement")
         {
             messageText = "Announcement";
+            QString announceCol = tags.value("msg-param-color").toString();
+            if (announceCol == "PRIMARY" || announceCol == "PURPLE" || announceCol == "")
+            {
+                announceFlag = MessageFlag::AnnouncePurple;
+            }
```

Comment:
Keep in mind that `msg-param-color` tag is not present at all times - invoking `.announce foo` from IRC completely skips it.
Perhaps you should try to use an iterator/check if the tag exists in the message before trying to 'stringify' it.
Nitpick, but why not name `announceCol` just `announceColor`.

I also don't like how handling of "PRIMARY" and empty string is packed together with the same condition with "PURPLE" - you shouldn't assume that purple will always be primary colour. First two mentioned by me cases should be treated as sort of 'fallback' in my opinion.

As something that's not necessarily up to you to figure out (but would be nice) is that this whole if-else chain could probably make use of newly added magic_enum dependency - and